camkes/parser/fdtQueryEngine.py

```diff
@@ -405,7 +406,12 @@ def _match_attr_dict(self, attr_dict):
 
     def query(self, attr):
         assert isinstance(attr, list)
-        return [self._match_attr_dict(attr_dict) for attr_dict in attr]
+
+        # Make a copy of each internal dict as _match_attr_dict mutates
```

Comment:
I assume `_match_attr_dict()` has due mutate the dict for other reasons then. Otherwise, wouldn't it be better to just not mutate the dict at all or add an explicit parameters that allows this, so the deep copy is not needed?

Reply:
It passes the dict through a couple different iterators that each potentially consume elements in the dict until they're all consumed so deleting entries is what it currently relies on to track progress. Not sure that rewriting would be that worthwhile because the variable the deep copy is being applied to is only going to consist of strings, numbers and lists/dicts of the same. No actual references to other objects.